src/greenpak/i2c.py

```diff
@@ -133,72 +174,112 @@ def __empty_rd(self, addr: int):
-                            try:
-                                self.bus.write_i2c_block_data(addr, regaddr, [data[1]])
-                            except Exception as e:
-                                print(f'{self.__class__.__name__}: ignoring exception "{e}" while erase page called.')
+                if start == 0xE3:  # An erase request, possibly
+                    caught_error = False
+                    ewhat= None
+                    try:
+                        self.bus.write_i2c_block_data(i2c_addr, start, data)
+                    except Exception as e:
+                        ewhat = e
+                        caught_error = True
+
+                    if caught_error:
+                        if self.trace_errors == True and ewhat != None:
+                            print(f'{self.__class__.__name__}: "{ewhat}" while trying to write to reg 0xE3...')
+                        import inspect
+                        stack = inspect.stack()
+                        for frame_info in stack:
+                            # ..but it can be anyone of 3 regions, and we don't know
+                            # at _this_ stack level where it is; upper driver has that nfo
+                            # So this is dirty as.
+                            if frame_info.function == '__erase_page':
+                                if self.trace_errors == True:
+                                    print(f'{self.__class__.__name__}: ignoring error while erase page called (device errata).')
+                                return True
+                    return False
             else:
-                # assume data is sent/written
-                regaddr = data[0]
-                regdata = data[1:]
-                bytesleft = len(regdata)
-                offset = 0
                 while bytesleft > 0:
                     xferlen = min(self.smbus2.I2C_SMBUS_BLOCK_MAX, bytesleft)
-                    self.bus.write_i2c_block_data(addr, regaddr, regdata[offset:offset + xferlen])
-                    offset += xferlen
+                    self.bus.write_i2c_block_data(i2c_addr, regaddr, data[offset:offset + xferlen])
+                    offset    += xferlen
                     bytesleft -= xferlen
-                    regaddr += xferlen
+                    regaddr   += xferlen
         except Exception as e:
-            if not silent:
-                str = "SMBus: while writing to addr 0x%02x: caught exception: %s" % (
-                    addr,
-                    e,
+            if self.trace_errors == True:
+                str = "SMBus: while writing i2c addr 0x%02x, reg 0x%02x: caught exception: %s" % (
+                    i2c_addr, regaddr, e
                 )
                 print(str)
             return False
+
         return True
 
-    @override
-    def read(
-        self, addr: int, byte_count: int, silent: bool = False
-    ) -> bytearray | None:
+    def write_read(self, addr: int, wreq: bytearray, rbytes: int,
```

Comment:
If this method is not used, please delete it. If it's used, make it private by adding '__' prefix.

Reply:
ok